main.go

```diff
@@ -608,3 +615,84 @@ func IndexCIEDE2000(pal color.Palette, orig color.Color) int {
 	}
 	return ret
 }
+
+// Color can convert itself to alpha-premultiplied 16-bits per channel RGBA.
+// The conversion may be lossy.
+type Color interface {
+	RGBA() (r, g, b, a uint32)
+}
+
+// Palette is a palette of colors.
+type Palette []Color
+
+// Find the brightest color
+func GreatestColor(r int, g int, b int) int {
+	if r > g && r > b {
+		return 1
+	}
+	if g > r && g > b {
+		return 2
+	}
+	if b > r && b > g {
+		return 3
+	}
+
+	return 0
+}
+
+// Index returns the index of the palette color closest to c in Euclidean
```

Comment:
By [GoDoc convention](https://blog.golang.org/godoc) doc comments start with the same name as the identifier they comment.

```suggestion
// IndexMult returns the index of the palette color closest to c in Euclidean
```